addons/event/models/event_registration.py

```diff
@@ -14,6 +17,20 @@ class EventRegistration(models.Model):
     _inherit = ['mail.thread', 'mail.activity.mixin']
     _order = 'id desc'
 
+    @api.model
+    def _get_random_token(self):
+        """Generate a string representation of a pseudo-random 8-byte number for barcode
+        generation.
+
+        A decimal serialisation is longer than a hexadecimal one *but* it
+        generates a more compact barcode (Code128C rather than Code128A).
+
+        Generate 8 bytes (64 bits) barcodes as 16 bytes barcodes are not
+        compatible with all scanners.
+         """
+        return str(int.from_bytes(os.urandom(8), 'little'))
+    barcode = fields.Char(default=_get_random_token, readonly=True, copy=False, index=True)
```

Comment:
hmm, that's a good question

Reply:
Maybe test this (e.g. change it to cause some collisions and observe the behavior), but I don't think it will behave "optimally" on its own. Maybe it'd be better to check if the randomly generated value is already in use inside this default function.

Reply:
@Pirols asked ORM team, they said that it will just raise an error, so when people see the error they can re-generate it again

Reply:
the `index=True` might be useless because there is an unique constraint for `unique(name)` which will create a unique btree index in PostgreSQL. Maybe an extra commit to remove it.